resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -751,12 +752,14 @@ private[spark] class Client(
       // by AM.
       Option(amKeytabFileName).foreach { k => props.setProperty(KEYTAB.key, k) }
       confStream.putNextEntry(new ZipEntry(SPARK_CONF_FILE))
-      val writer = new OutputStreamWriter(confStream, StandardCharsets.UTF_8)
+      writer = new OutputStreamWriter(confStream, StandardCharsets.UTF_8)
```

Comment:
Closing confStream was already sufficient here.  I think this is cleaner with a nested try, or if you like, just closing this object after flushing